packages/aws-cdk-lib/aws-ecs/lib/amis.ts

```diff
@@ -97,6 +97,14 @@ export interface EcsOptimizedAmiProps {
    * @default false
    */
   readonly cachedInContext?: boolean;
+
+  /**
+   * Kernel version used.
+   * Only supported for Amazon Linux 2.
+   *
+   * @default none, uses the recommended kernel version
+   */
+  readonly kernel?: string;
```

Comment:
As @TheRealAmazonKendra said, this is a deprecated interface, so we shouldn't be adding a new property to this...except, there's a note that says `remove @deprecated` above in V2. @rix0rrr I'm thinking we should un-deprecate this property, WDYT? 

This should not be a string value, it should be an enum. We should not force users to call the `.toString()` on the enum, they should just pass it directly. 

The enum provides type discoverability; if the `kernel` property is only a string type, there's nothing the IDE can do to tell you that you should really be passing the result of a `toString()` call. Instead, with the enum, it tells you what type is expected, so the user discovers what they need to pass much more easily.

Reply:
@comcalvi Thanks for the review.

1) Regarding the deprecated interface:
> This is a deprecated interface, so we shouldn't be adding a new property to this.

There was some back and forth on this point. I opted for this workaround to avoid introducing a breaking change, which is deemed unacceptable as per [this comment](https://github.com/aws/aws-cdk/pull/29809#discussion_r1568061197).

> So, it's true that you can get around our linter by adding that, but this is a stable module, so we can't actually accept a breaking change here.

2) On using an enum instead of a string:
> This should not be a string value, it should be an enum. We should not force users to call the .toString() on the enum.

I have updated the implementation to pass the enum directly. However, I didn't foresee this option being passed by a user. The constructor is private, and only the static methods are instantiating the class. Therefore, there was no way for any user (except through the static methods) to call the constructor directly.